src/index.ts

```diff
@@ -1,81 +1,155 @@
+                kaiEvents.emit('default','Invalid response')
+                break;
+        };
+    }else{
+        kaiEvents.emit(response.type,response.error)
+    }
 });
 
-ws.on('message', (data) => {
-	var input = JSON.parse(data.toString());
-	KaiEvents.emit('data', input);
+//TODO:-
+export function setCapabilities(kaiId:number|"default"|"defaultLeft"|"defaultRight",capabilitiesArr:string[]){ 
+        var request:object = {
+          type: 'setCapabilities',
+          kaiId: kaiId
+        }
+        var obj = {...request}
+        const req = (this.obj as any)
+        capabilitiesArr.forEach(element => {
+          req[element] = true
+        });
+        
+
+        ws.on('open',function(){
+            ws.send(JSON.stringify(request));
+        });
+}
 
-	if(input.success != true) {
-		KaiEvents.emit('error', {
-			errorCode: input.errorCode,
-			error: input.error,
-			message: input.message
-		});
-		return;
-	}
+//NOTE : getCapbilities isn't working right now
+export function getCapabilities(kaiId:number|"default"|"defaultLeft"|"defaultRight"="default"){
+    let request:sdk.GetCapabilitiesRequest={
+        type:'getCapabilities',
+        kaiId: kaiId
+    };
+    ws.on('open',function(){
+        ws.send(JSON.stringify(request));
+    });
+};
 
-	switch(input.type) {
-		case "gestureData":
-			KaiEvents.emit('gestureData', input.gesture);
-			break;
-		// TODO other types of data
-	}
-});
+export function incomingData(){
+    let request={
+        type:'incomingData'
+    };
+    ws.on('open',function(){
+        ws.send(JSON.stringify(request));
+    });
+}
+export function fingerCalibration(kaiId:number|"default"|"defaultLeft"|"defaultRight"="default"){
+    let request:sdk.FingerCalibrationRequest={
+        type:'fingerCalibration',
+        kaiId:kaiId
+    };
+    ws.on('open',function(){
+        ws.send(JSON.stringify(request));
+    });
+};
+
+export function listConnectedKais(){
+    let request:sdk.ListConnectedKaisRequest={
+        type:'listConnectedKais'
+    };
+    ws.on('open',function(){
+        ws.send(JSON.stringify(request))
+    });
+};
 
-export function connect() {
-	ws = new WebSocket('ws://localhost:2203');
+export function getKaiData(kaiId:number|"default"|"defaultLeft"|"defaultRight"="default"){
+    let request:sdk.GetKaiDataRequest = {
+        type:'getKaiData',
+        kaiId:kaiId
+    }   
+    ws.on('open',function(){
+        ws.send(JSON.stringify(request))
+    });    
 }
 
-export function subscribe(...capabilities: KaiCapabilities[]) {
-	return new Promise((resolve, reject) => {
-		let json: any = {
-			type: 'setCapabilities'
-		};
-		capabilities.forEach(capability => {
-			json[toCamelCase(capability)] = true;
-		});
-		if(ws) {
-			resolve();
-		}
-		ws.send(json, err => {
-			if(err)
-				reject(err);
-			else
-				resolve();
-		});
-	});
+export function switchHand(kaiId:number|"default"|"defaultLeft"|"defaultRight"="default",hand:"left"|"right"="left"){
```

Comment:
The type here for the hand parameter as well, import it from the model. Do not define types inside any other file or in this way, it is very inconsistent, always import it from the model.